ptf/sai_base_test.py

```diff
@@ -0,0 +1,1176 @@
+from ptf.base_tests import BaseTest
+
+from thrift.transport import TSocket
+from thrift.transport import TTransport
+from thrift.protocol import TBinaryProtocol
+
+from sai_thrift import sai_rpc
+
+from sai_adapter import *
+
+ROUTER_MAC = '00:77:66:55:44:00'
+THRIFT_PORT = 9092
+
+
+class ThriftInterface(BaseTest):
+    """
+    Get and format a port map, retrieve test params, and create an RPC client
+    """
+    def setUp(self):
+        super(ThriftInterface, self).setUp()
+
+        self.interface_to_front_mapping = {}
+        self.port_map_loaded = False
+        self.transport = None
+
+        self.test_params = testutils.test_params_get()
+        self.loadPortMap()
+        self.createRpcClient()
+
+    def tearDown(self):
+        self.transport.close()
+
+        super(ThriftInterface, self).tearDown()
+
+    def loadPortMap(self):
+        """
+        Get and format port_map
+
+        port_map_file is a port map with following lines format:
+        [test_port_no]@[device_port_name]
+        e.g.:
+             0@Veth1
+             1@Veth2
+             2@Veth3  ...
+        """
+        if self.port_map_loaded:
+            print("port_map already loaded")
+            return
+
+        if "port_map" in self.test_params:
+            user_input = self.test_params['port_map']
+            splitted_map = user_input.split(",")
+            for item in splitted_map:
+                iface_front_pair = item.split("@")
+                self.interface_to_front_mapping[iface_front_pair[0]] =  \
+                    iface_front_pair[1]
+        elif "port_map_file" in self.test_params:
+            user_input = self.test_params['port_map_file']
+            with open(user_input, 'r') as map_file:
+                for line in map_file:
+                    if (line and
+                        (line[0] == '#' or line[0] == ';' or line[0] == '/')):
+                        continue
+                    iface_front_pair = line.split("@")
+                    self.interface_to_front_mapping[iface_front_pair[0]] =  \
+                        iface_front_pair[1].strip()
+
+        self.port_map_loaded = True
+
+    def createRpcClient(self):
+        """
+        Set up thrift client and contact RPC server
+        """
+
+        if 'thrift_server' in self.test_params:
+            server = self.test_params['thrift_server']
+        else:
+            server = 'localhost'
+
+        self.transport = TSocket.TSocket(server, THRIFT_PORT)
+        self.transport = TTransport.TBufferedTransport(self.transport)
+        self.protocol = TBinaryProtocol.TBinaryProtocol(self.transport)
+
+        self.client = sai_rpc.Client(self.protocol)
+        self.transport.open()
+
+
+class ThriftInterfaceDataPlane(ThriftInterface):
+    """
+    Sets up the thrift interface and dataplane
+    """
+    def setUp(self):
+        super(ThriftInterfaceDataPlane, self).setUp()
+
+        self.dataplane = ptf.dataplane_instance
+        self.removeCpuPort()
```

Comment:
Please remove this also. We do not want to expose cpu ports externally

Reply:
sorry, it was left by mistake, done